src/backend/joanie/core/utils/newsletter/subscription.py

```diff
@@ -12,20 +12,18 @@
 
 
 @app.task
-def set_commercial_newsletter_subscription(user_dict):
+def set_commercial_newsletter_subscription(user_id):
     """
     Set the newsletter subscription for the user.
     """
+    User = apps.get_model("core", "User")  # pylint: disable=invalid-name
+    user_dict = User.objects.get(id=user_id).to_dict()
```

Comment:
As we know there is a race condition on this part on the code, can we wrap it with a user does not exist exception ?